generators/go-v2/codegen/src/ast/Func.ts

```diff
@@ -0,0 +1,45 @@
+import { GoTypeReference } from "./GoTypeReference";
+import { Comment } from "./Comment";
+import { CodeBlock } from "./CodeBlock";
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+import { Method } from "./Method";
+import { Parameter } from "./Parameter";
+import { Type } from "./Type";
+
+export declare namespace Func {
+    type Args = Omit<Method.Args, "typeReference">;
+}
+
+export class Func extends AstNode {
+    private func: Method;
+
+    constructor({ name, parameters, return_, body, docs }: Func.Args) {
+        super();
+        this.func = new Method({ name, parameters, return_, body, docs });
+    }
+
+    public get name(): string {
+        return this.func.name;
+    }
+
+    public get parameters(): Parameter[] {
+        return this.func.parameters;
+    }
+
+    public get return_(): Type[] {
+        return this.func.return_;
+    }
+
+    public get body(): CodeBlock | undefined {
+        return this.func.body;
+    }
+
+    public get docs(): string | undefined {
+        return this.func.docs;
+    }
```

Comment:
this is clean, feel like we should do this in other places as opposed to making attributes readonly

Reply:
Glad you noticed this! I was exploring options here when I asked myself, "What's the best way to restrict the scope of _this_ class, but delegate to a common structure?" Happy to use this pattern elsewhere as we go.